pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -945,11 +945,11 @@ func (m *kubeGenericRuntimeManager) doBackOff(pod *v1.Pod, container *v1.Contain
 	}
 
 	klog.V(3).InfoS("Checking backoff for container in pod", "containerName", container.Name, "pod", klog.KObj(pod))
-	// Use the finished time of the latest exited container as the start point to calculate whether to do back-off.
-	ts := cStatus.FinishedAt
+
 	// backOff requires a unique key to identify the container.
 	key := getStableKey(pod, container)
-	if backOff.IsInBackOffSince(key, ts) {
+	now := time.Now()
+	if backOff.IsInBackOffSinceUpdate(key, now) {
```

Comment:
This doesn't seem right to me. We reset the time every run through this function, so won't that prevent an exponential back off? According to the output in the first comment, the restart interval is linear/fixed now, which I believe is a change in behaviour.

The root issue seems to be that there's some sort of race described in #99697 resulting in two restarts. I'm not sure how this change addresses that.

/hold

Reply:
If you compare `backOff.IsInBackOffSince(key, ts)`
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/client-go/util/flowcontrol/backoff.go#L92-L103
with `backOff.IsInBackOffSinceUpdate(key, now)`
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/client-go/util/flowcontrol/backoff.go#L106-L117

You can see the only difference is on the condition `hasExpired`, so does it really changes the exponential backoff?
This might eventually come from the change that I requested when calling `backOff.Next(key, now)`.

I think we should be asking ourselves why there is a bug on the condition described above... also this func in only called from here which would make it easier to fix without side effects.

Also this calls for more investigation:

> Maybe not, If the time passed to IsInBackOffSinceUpdate does not change, and Next is not called, then the result returned by IsInBackOffSinceUpdate will not change. FinishedAt of the container remains unchanged until the next container. so it has been in the same time window.

Which is what we see happening...